### PR TITLE
+act #17274 make ForkJoinPool asyncMode configurable (for validation)

### PR DESCRIPTION
**Pull request by patriknw**
(cherry picked from commit 05f156bdc0eae121aa122028a582084bc4bb22dc)

### CODE REVIEW

**Comment by patriknw** [comment]
refs #17274

File: akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala [hunk 1]
```diff
@@ -424,12 +430,20 @@ class ForkJoinExecutorConfigurator(config: Config, prerequisites: DispatcherPrer
         m.withName(m.name + "-" + id)
       case other ⇒ other
     }
+
+    val asyncMode = config.getString("task-peeking-mode") match {
+      case "FIFO"      ⇒ true
+      case "LIFO"      ⇒ false
+      case unsupported ⇒ throw new IllegalArgumentException(s"""Cannot instantiate ForkJoinExecutorServiceFactory. "task-peeking-mode" in "fork-join-executor" section could only set to "FIFO" or "LILO".""")
```
Comment (richdougherty) [inline akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala]: I was just testing this and I noticed a typo in this error message which might confuse users: `LILO` → `LIFO`.

Comment (hepin1989) [inline akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala]: sorry for that,Will update now.

Comment (hepin1989) [inline akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala]: refs https://github.com/akka/akka/issues/17482

**Pull request merged**